pkg/receive/writer.go

```diff
@@ -104,6 +135,18 @@ func (r *Writer) Write(ctx context.Context, tenantID string, wreq *prompb.WriteR
 		level.Warn(r.logger).Log("msg", "Error on ingesting samples that are too old or are too far into the future", "numDropped", numOutOfBounds)
 		errs.Add(errors.Wrapf(storage.ErrOutOfBounds, "add %d samples", numOutOfBounds))
 	}
+	if numExemplarsOutOfOrder > 0 {
+		level.Warn(r.logger).Log("msg", "Error on ingesting out-of-order exemplars", "numDropped", numExemplarsOutOfOrder)
```

Comment:
It's a level warn, but the logger msg says it's an error. I would make the message more generic and let log level handle the prefix, like: "warn: exemplars out of order" or something like that.

Reply:
I was following what we do for samples. I'd like other's opinion here as well to either switch to error, put warn there instead of error, or keep it as it.

Reply:
I think we are not too strict here, so it's fine to say error if it's only a warning about the not serious error.

Reply:
If we handle the returned error on upper levels (this could be logging as well), we don't need to log/handle it twice. I see that you've just followed the existing structure so it seems ok for now.

Reply:
hm.. what I don't like is duplicate error handling. We both add error and warn error... Wonder if we could just handle that on caller side instead. 🤔

Reply:
Yeah, this is a bit icky but as Kemal noted I was just following the existing structure we use for sample. I think I can try refactoring this in a follow up PR if that's ok.